docs/proposal/2022-06-14-configuration-bindings.md

```diff
@@ -0,0 +1,647 @@
+
+    ```yaml
+    communications: # it' is a list so it allows to have e.g multiple Elasticsearch configurations
+      - name: tenant-b-workspace
+        slack:
+          token: 'SLACK_API_TOKEN'
+          # ...trimmed...
+
+          # describe channel bindings
+          channels:
+            - name: "#team-a"
+              bindings:
+                sources:
+                  - "nodes-errors"
+                  - "deprecated-api"
+                executors:
+                  - "kubectl-read-only"
+                  - "helm-full-access"
+        elasticsearch:
+          server: 'ELASTICSEARCH_ADDRESS'
+          # ...trimmed...
+
+          # ELS index bindings
+          index:
+            - destination:
+                name: network-errors
+                type: botkube-event
+                shards: 1
+              bindings:
+                sources:
+                  - "nodes-errors"
+                  - "depreacted-api"
+                # executors - not allowed in this case, ES is "sink" only.
+    ```
+    </td>
+    </tr>
+    </table>
+
+## Use cases
+
+This section describes example configurations that enable the requested use-cases.
+
+### Route BotKube notifications to individual channels
+
+With presented configuration:
+- nodes errors are sent to a `#nodes` channel,
+- while networks errors are sent to the `#network` channel.
+
+**Communicators**
+
+```yaml
+communications: # allows to have multiple slacks, or ES configurations
+  - name: tenant-workspace
+    slack:
+      token: 'SLACK_API_TOKEN'
+      # customized notifications
+      channels:
+        - name: "#nodes"
+          bindings:
+            sources:
+              - "nodes-errors"
+        - name: "#network"
+          bindings:
+            sources:
+              - "network-errors"
+```
+
+**Sources**
+
+```yaml
+Sources:
+ - name: nodes-errors # name used for bindings
+   kubernetes:
+     resources:
+       - name: v1/nodes
+         namespaces:
+           include:
+             - @all
+         events:
+           - error
+  - name: network-errors
+    kubernetes:
+      namespace:
+        include: ["@all"]
+      resources:
+        - name: v1/pods
+          namespaces: # override the top level Namespace
+            include:
+              - istio-system
+          events:
+            - error
+        - name: v1/services
+          events:
+            - error
+        - name: networking.istio.io/v1alpha3/DestinationRules
+          events:
+            - error
+        - name: networking.istio.io/v1alpha3/VirtualServices
+          events:
+            - error
+```
+
+### Route notifications to a given channel based on the Kubernetes Namespace
+
+> **Note**
+>
+> Currently, you can send notification to non-default channel using [annotation](https://www.botkube.io/usage/#send-notification-to-non-default-channel).
+However, you need to apply `botkube.io/channel: <channel_name>` to each K8s object (Pods, Services, etc.) which is cumbersome.
```

Comment:
so you plan to move `botkube.io/channel` to namespace level, right?
what about `botkube.io/disable` ?
I suggest to keep the disable annotation on resource level, but extend it to namespace level as well.

Reply:
It's fine like you described. But what about the other annotation `botkube.io/disable`?